lib/kadai-core/src/main/java/io/kadai/task/internal/TaskDistributor.java

```diff
@@ -0,0 +1,277 @@
+  }
+
+  public BulkOperationResults<String, KadaiException> distribute(
+      String sourceWorkbasketId,
+      List<String> destinationWorkbasketIds,
+      String distributionStrategyName,
+      Map<String, Object> additionalInformation)
+      throws NotAuthorizedOnWorkbasketException,
+          WorkbasketNotFoundException,
+          TaskNotFoundException {
+
+    return distributeTasks(
+        sourceWorkbasketId,
+        null,
+        distributionStrategyName,
+        destinationWorkbasketIds,
+        additionalInformation);
+  }
+
+  public BulkOperationResults<String, KadaiException> distribute(String sourceWorkbasketId)
+      throws NotAuthorizedOnWorkbasketException,
+          WorkbasketNotFoundException,
+          TaskNotFoundException {
+
+    return distributeTasks(sourceWorkbasketId, null, null, null, null);
+  }
+
+  public BulkOperationResults<String, KadaiException> distribute(
+      String sourceWorkbasketId, List<String> taskIds)
+      throws NotAuthorizedOnWorkbasketException,
+          WorkbasketNotFoundException,
+          TaskNotFoundException {
+
+    return distributeTasks(sourceWorkbasketId, taskIds, null, null, null);
+  }
+
+  public BulkOperationResults<String, KadaiException> distributeWithStrategy(
+      String sourceWorkbasketId,
+      List<String> taskIds,
+      String distributionStrategyName,
+      Map<String, Object> additionalInformation)
+      throws NotAuthorizedOnWorkbasketException,
+          WorkbasketNotFoundException,
+          TaskNotFoundException {
+
+    return distributeTasks(
+        sourceWorkbasketId, taskIds, distributionStrategyName, null, additionalInformation);
+  }
+
+  public BulkOperationResults<String, KadaiException> distributeWithStrategy(
+      String sourceWorkbasketId,
+      String distributionStrategyName,
+      Map<String, Object> additionalInformation)
+      throws NotAuthorizedOnWorkbasketException,
+          WorkbasketNotFoundException,
+          TaskNotFoundException {
+
+    return distributeTasks(
+        sourceWorkbasketId, null, distributionStrategyName, null, additionalInformation);
+  }
+
+  public BulkOperationResults<String, KadaiException> distributeWithDestinations(
+      String sourceWorkbasketId, List<String> destinationWorkbasketIds)
+      throws NotAuthorizedOnWorkbasketException,
+          WorkbasketNotFoundException,
+          TaskNotFoundException {
+
+    return distributeTasks(sourceWorkbasketId, null, null, destinationWorkbasketIds, null);
+  }
+
+  public BulkOperationResults<String, KadaiException> distributeWithDestinations(
+      String sourceWorkbasketId, List<String> taskIds, List<String> destinationWorkbasketIds)
+      throws NotAuthorizedOnWorkbasketException,
+          WorkbasketNotFoundException,
+          TaskNotFoundException {
+
+    return distributeTasks(sourceWorkbasketId, taskIds, null, destinationWorkbasketIds, null);
+  }
+
+  private BulkOperationResults<String, KadaiException> distributeTasks(
```

Comment:
this looks fine now. However the method is doing a lot of things. 

(checking authorization, but this is a 1-liner)
-resolving and validating TaskIds
-resolving and validating destinationWorkbasketIds
-retrieving the transferFlag, partitioning the tasks for transfer and finally transferring

how would you feel about refactoring some of these into smaller methods for better readability? :) This is just a suggestion and I am willing to approve if you don't like the idea